### PR TITLE
Add definition for release manager

### LINKED ISSUE #4942: Add definition for release manager
As discussed during the March 2nd SIG meeting, we don't have a definition of release manager in the release documentation. @jpkrohling volunteered to add one

### PR DESCRIPTION
**Pull request by jpkrohling**
Closes #4942
    
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Review by codeboten — APPROVED** [review]

File: docs/release.md [hunk 1]
```diff
@@ -15,6 +15,12 @@ We release both core and contrib collectors with the same versions where the con
 * [open-telemetry/opentelemetry-collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib)
 * [open-telemetry/opentelemetry-collector-releases](https://github.com/open-telemetry/opentelemetry-collector-releases)
 
+## Release manager
+
+A release manager is the person responsible for a specific release. While the manager might request help from other folks, they are the ultimate responsible for the success of a release.
```
Comment (Aneurysm9) [inline docs/release.md:20]:
```suggestion
A release manager is the person responsible for a specific release. While the manager might request help from other folks, they are ultimately responsible for the success of a release.
```

File: README.md [hunk 2]
```diff
@@ -144,6 +144,8 @@ Here is a list of community roles with current and previous members:
    - [Paulo Janotti](https://github.com/pjanotti), Splunk
 
 Learn more about roles in [Community membership](https://github.com/open-telemetry/community/blob/main/community-membership.md).
+In addition to what is described at the organization-level, the SIG Collector requires all core approvers to take part in rotating
```
Comment (Aneurysm9) [inline README.md:147]: Does this assume that all core approvers are also contrib approvers?  Should that be made explicit?

Comment (bogdandrutu) [reply README.md:147]: I think it assumes, but I don't think it should be.

Comment (jpkrohling) [reply README.md:147]: The idea is that a specific group should be responsible for the releases, and the people belonging to this group should rotate doing this work. If a core approver isn't an approver on contrib, they should request assistance from a cotnrib approver. I'll make this explicit.

If you think another group should be responsible for the release, let me know.

**Pull request merged**